### PR TITLE
Migrate Profile mutation

### PR DESCRIPTION
**Pull request by pixelfact**

### CODE REVIEW

File: src/App/Stacks/ContributorProfileSidePanel/ReadOnlyView/IntroSection.tsx [hunk 1]
```diff
@@ -19,23 +19,24 @@ import { cn } from "src/utils/cn";
...
 import { components } from "src/__generated/api";
-import { OwnUserProfileDetailsFragment, UserProfileFragment } from "src/__generated/graphql";
+import { UseGetMyProfileInfoResponse } from "src/api/me/queries";
+import { calculateUserCompletionScore } from "src/utils/calculateCompletionScore";
+import CompletionBar from "src/components/CompletionBar";
 
 type Props = {
   profile: Profile;
-  gqlProfile?: UserProfileFragment & OwnUserProfileDetailsFragment; // use this for the completion score, should be revamp when we revamp the edit profile
+  myProfile?: UseGetMyProfileInfoResponse;
```
Comment (haydencleary) [inline src/App/Stacks/ContributorProfileSidePanel/ReadOnlyView/IntroSection.tsx:31]: Try to merge these props, we shouldn't need two data entries

File: src/api/me/mutations.ts [hunk 2]
```diff
@@ -69,4 +69,26 @@ const useApplyProject = ({
...
-export default { useAcceptProjectLeaderInvitation, useClaimProject, usePayoutInfo, useApplyProject, useUpdateMe };
+export type UseUpdateProfileBody = components["schemas"]["UserProfileRequest"];
+export type UseUpdateProfileResponse = components["schemas"]["PrivateUserProfileResponse"];
+
+const useUpdateProfile = ({ options = {} }: UseMutationProps<UseUpdateProfileResponse, UseUpdateProfileBody>) => {
+  return useBaseMutation<UseUpdateProfileBody, UseUpdateProfileResponse>({
+    resourcePath: API_PATH.ME_PROFILE,
+    invalidatesTags: [
+      { queryKey: MeApi.tags.all, exact: false },
+      { queryKey: ["resftullProfile"], exact: false },
```
Comment (haydencleary) [inline src/api/me/mutations.ts:80]: Check `resftullProfile`

File: src/utils/calculateCompletionScore.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+function calculateUserCompletionScore(userProfile: UseGetMyProfileInfoResponse) {
+  const scoreByExistence = (fieldName: keyof typeof userProfile, score: number) => (userProfile[fieldName] ? score : 0);
+
+  const scoreContact = (channel: Channel, score: number) =>
+    userProfile.contacts?.find(contact => contact.channel === channel && contact.contact) ? score : 0;
+
+  const scoreTechnologies = (score: number) => (Object.keys(userProfile.technologies ?? {}).length ? score : 0);
+
+  return (
+    scoreByExistence("avatarUrl", 5) +
```
Comment (haydencleary) [inline src/utils/calculateCompletionScore.ts:16]: Make a dictionary to store each element's score

File: src/App/Stacks/ContributorProfileSidePanel/EditView/index.tsx [hunk 4]
```diff
@@ -29,71 +21,64 @@ import { useState } from "react";
...
-
   const formMethods = useForm<UserProfileInfo>({
-    defaultValues: fromFragment(profile),
+    defaultValues: fromFragment(myProfile),
     mode: "onChange",
   });
   const { handleSubmit, formState, control, getValues } = formMethods;
   const { isDirty, isValid } = formState;
-  const [completionScore, setCompletionScore] = useState(profile.completionScore);
+  const [completionScore, setCompletionScore] = useState(calculateUserCompletionScore(myProfile));
```
Comment (alexbeno) [inline src/App/Stacks/ContributorProfileSidePanel/EditView/index.tsx:45]: The use state is not sync if myProfile change it's ok ?

Comment (alexbeno) [reply src/App/Stacks/ContributorProfileSidePanel/EditView/index.tsx:45]: Ok got it !

**Review by haydencleary — APPROVED** [review]

**Pull request merged**